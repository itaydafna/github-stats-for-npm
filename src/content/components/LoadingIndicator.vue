<template>
  <div id="container">
    <div id="loading">
      <div class="line inc"/>
      <div class="line dec"/>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {};
  },
};
</script>

<style scoped>
#container {
  position: relative;
}
#loading {
  position: absolute;
  top: -10px;
  width: 80px;
  height: 5px;
  overflow-x: hidden;
}

.line {
  position: absolute;
  background: #000;
  height: 5px;
}
.inc {
  animation: increase 1s infinite;
}
.dec {
  animation: decrease 1s 0.25s infinite;
}

@keyframes increase {
  from {
    left: -5%;
    width: 5%;
  }
  to {
    left: 130%;
    width: 100%;
  }
}
@keyframes decrease {
  from {
    left: -80%;
    width: 80%;
  }
  to {
    left: 110%;
    width: 10%;
  }
}
</style>
