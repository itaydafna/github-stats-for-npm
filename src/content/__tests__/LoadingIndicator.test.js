import { mount } from '@vue/test-utils';
import LoadingIndicator from '../components/LoadingIndicator';

describe('LoadingIndicator', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(LoadingIndicator);
  });
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  it('renders a loading indicator', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
