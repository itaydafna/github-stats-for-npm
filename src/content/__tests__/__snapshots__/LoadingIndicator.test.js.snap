// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator renders a loading indicator 1`] = `
<div
  id="container"
>
  <div
    id="loading"
  >
    <div
      class="line inc"
    />
     
    <div
      class="line dec"
    />
  </div>
</div>
`;
